src/src/SessionStore.cpp

```diff
@@ -0,0 +1,206 @@
+/*
+** EPITECH PROJECT, 2022
+** SessionStore.cpp
+** File description:
+** SessionStore.cpp
+*/
+
+
+#include "Polymorph/Network/SessionStore.hpp"
+#include "authorizationKey.hpp"
+#include "Polymorph/Network/exceptions/UnknownAuthorizationKeyException.hpp"
+#include "Polymorph/Network/exceptions/UnauthorizedException.hpp"
+#include "Polymorph/Network/exceptions/AlreadyRegisteredException.hpp"
+#include "Polymorph/Network/exceptions/UnknownSessionException.hpp"
+
+void polymorph::network::SessionStore::registerAuthoredClient(asio::ip::udp::endpoint endpoint,
+        polymorph::network::SessionId sessionId, polymorph::network::AuthorizationKey key)
+{
+    try {
+        std::unique_lock<std::mutex> lock(_udpSessionsAuthorizationKeysMutex);
+        auto verifyKey = _udpSessionsAuthorizationKeys.at(sessionId);
+        lock.unlock();
+        if (authorizationKey::areSame(verifyKey, key)) {
+            _udpSessions[endpoint] = sessionId;
+        } else {
+            throw exceptions::UnauthorizedException("InvalidAuthorization key");
+        }
+    } catch (std::out_of_range &e) {
+        throw exceptions::UnknownAuthorizationKeyException("Unknown authorization key");
+    }
+}
+
+polymorph::network::SessionId polymorph::network::SessionStore::registerClient(asio::ip::udp::endpoint endpoint)
+{
+    try {
+        std::lock_guard<std::mutex> lock(_udpSessionsMutex);
+        return _udpSessions.at(endpoint);
+    } catch (std::out_of_range &e) {
+        auto sId = _findAvailableUdpSessionId();
+        std::lock_guard<std::mutex> lock(_udpSessionsMutex);
+        _udpSessions.emplace(endpoint, sId);
+        return sId;
+    }
+}
+
+void polymorph::network::SessionStore::registerAuthoredClient(asio::ip::tcp::endpoint endpoint,
+                                                              polymorph::network::SessionId sessionId, polymorph::network::AuthorizationKey key)
+{
+    try {
+        std::unique_lock<std::mutex> lock(_tcpSessionsAuthorizationKeysMutex);
+        auto verifyKey = _tcpSessionsAuthorizationKeys.at(sessionId);
+        lock.unlock();
+        if (authorizationKey::areSame(verifyKey, key)) {
+            _tcpSessions[endpoint] = sessionId;
+        } else {
+            throw exceptions::UnauthorizedException("InvalidAuthorization key");
+        }
+    } catch (std::out_of_range &e) {
+        throw exceptions::UnknownAuthorizationKeyException("Unknown authorization key");
+    }
+}
+
+polymorph::network::SessionId polymorph::network::SessionStore::registerClient(asio::ip::tcp::endpoint endpoint)
+{
+    auto sId = _findAvailableTcpSessionId();
+    std::lock_guard<std::mutex> lock(_tcpSessionsMutex);
+
+    try {
```

Comment:
Using contains or any conditional presence will be better than an entire try catch block